File: sorting/C++/mergeSort.c
#include<stdio.h>

// void merge(){

// }

void mergeSort(int a[], int size){
    int c,i, l[50], r[50], mid = size/2;
    if(size < 1) return;
    for(c=i=0;c<mid;c++){
        l[i] = a[c];
        i++;
    } for(c=mid,i=0;c<size;c++){
        r[i] = a[c];
        i++;
    }
    mergeSort(l,mid);
    mergeSort(r,size-mid);
    merge(a,l,r);
}

void main(){
    int n,i;
    int a[100];
    scanf("%d",&n);//size of list
    for(i=0;i<n;i++) scanf("%d",&a[i]);
    mergeSort(a,n);
    printf("Sorted List -->\n");
    for(i=0;i<n;i++) printf("%d",a[i]);
}

File: sorting/python3/mergeSort.py
def mergeSort(data):
    if len(data) > 1:
        mid = len(data) //2
        leftPart = data[:mid]
        rightPart = data[mid:]
        mergeSort(leftPart)
        mergeSort(rightPart)
        i = j = k = 0
        while i<len(leftPart) and j <len(rightPart):
            if leftPart[i] < rightPart[j]:
                data[k] = leftPart[i]
                i += 1
            else:
                data[k] = rightPart[j]
                j += 1
            k += 1
        while i < len(leftPart):
            data[k] = leftPart[i]
            i += 1
            k += 1
        while j < len(rightPart):
            data[k] = rightPart[j]
            j += 1
            k += 1
if __name__ == "__main__":
    n = int(input("Enter number of data: "))
    data = [int(input()) for x in range(n)]
    mergeSort(data)
    print(data)